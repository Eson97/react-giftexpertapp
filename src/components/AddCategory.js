import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = (({ setCategories }) => {

    const [inputValue, setInputValue] = useState('Nueva categorÃ­a');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }

    const handleFocus = () => {
        setInputValue('');
    }

    const handleBlur = () =>{
        setInputValue('Nueva categoria');
    }

    return (
        <form onSubmit={handleSubmit} onFocus={handleFocus} onBlur={handleBlur}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )

})

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
