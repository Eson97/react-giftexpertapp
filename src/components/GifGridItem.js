import React from 'react'



export const GifGridItem = ({title,url}) => {
    
    return (
        <div className="card animate__animated animate__fadeInLeft">
            <img src={url} alt={title}/>
            <p>{title}</p>
            <iframe title="share-button" src={`https://www.facebook.com/plugins/share_button.php?href=${url}&layout=button&size=large&width=103&height=28&appId`} width="103" height="28" style={{border:"none",overflow:"hidden",margin:"10px"}} scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>
        </div>
    )
}
