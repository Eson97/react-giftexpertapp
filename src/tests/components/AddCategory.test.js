import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('test on <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;
    // const wrapper= shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should show correctly', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('should change textbox', () => {

        const value = 'hola mundo'
        
        //POR ALGUNA RAZON SI LO LLAMO DESDE VARIABLE NO FUNCIONA
        // const input = wrapper.find('input');
        
        // input.simulate('change', { target: { value } });

        // expect(input.prop('value')).toBe(value);
        
        wrapper.find('input').simulate('change', { target: { value } });

        expect(wrapper.find('input').prop('value')).toBe(value);

    })

    test('should NOT post info with onSubmit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();

    })

    test('should call setCatgories and clear textbox', () => {
        
        const value = 'hola mundo';
        
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');
    })


})
