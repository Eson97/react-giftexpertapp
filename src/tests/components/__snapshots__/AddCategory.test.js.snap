// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <AddCategory /> should show correctly 1`] = `
<form
  onBlur={[Function]}
  onFocus={[Function]}
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
